src/utils/date.ts

```diff
@@ -1,12 +1,27 @@
 import { DateLocaleData } from "../components/date-picker/utils";
-import { numberStringFormatter } from "./locale";
+import { numberStringFormatter, locales } from "./locale";
 
 export interface HoverRange {
   focused: "end" | "start";
   start: Date;
   end: Date;
 }
 
+const getUniqueDateSeparators = () => {
+  const uniqueDateSeparators = new Set();
+  const dummyDate = new Date(2020, 4, 20);
+  locales.forEach((locale) => {
+    const dateTimeFormat = new Intl.DateTimeFormat(locale, { year: "numeric", month: "numeric", day: "numeric" });
+
+    const parts = dateTimeFormat.formatToParts(dummyDate);
+    const separator = parts.filter((part) => part.type === "literal")[0]?.value;
+    uniqueDateSeparators.add(separator);
+  });
+  return [...uniqueDateSeparators];
+};
+
+export const dateSeparators = getUniqueDateSeparators();
```

Comment:
Do we expect this to grow with new locales added? I guess this is safer if locales change.

Reply:
If we add support for additional locales they should be added to the array I'm iterating through, so any additional separators would be represented here. Or am I misunderstanding?

Reply:
Yeah that's correct. I was just wondering if we expect the list to grow or if we could just generate the const once and not have it dynamic like this.

Reply:
Maybe we should add a spec test for this?

Reply:
Yeah generating the const once was what I originally did, which came out to `["/", ".", "-"]` after trimming. I'm fine with either. Realistically this code shouldn't make our break our localization lol. Something in the component's lifecycle needs to be fixed, this is just a temporary bandaid.